web-app/prisma/schema.prisma

```diff
@@ -353,23 +354,98 @@ enum AgentListType {
   FAVORITE
 }
 
+model CreditTransaction {
+  id            String                @id @default(cuid())
+  createdAt     DateTime              @default(now())
+  updatedAt     DateTime              @updatedAt
+  amount        BigInt //positive is always a credit topup and negative is always a credit spend
+  includedFee   BigInt
+  note          String?
+  noteKey       String?
+  errorNote     String?
+  errorNoteKey  String?
+  type          CreditTransactionType
+  userId        String
+  user          User                  @relation(fields: [userId], references: [id], onDelete: Cascade)
+  fiatPayment   FiatPayment?          @relation(fields: [fiatPaymentId], references: [id])
+  fiatPaymentId String?               @unique
+
+  status      CreditTransactionStatus @default(PENDING)
+  job         Job?                    @relation("JobCost")
+  refundedJob Job?                    @relation("JobRefund")
+}
+
+enum CreditTransactionStatus {
+  PENDING
+  SUCCEEDED
+  FAILED
+}
+
+enum CreditTransactionType {
+  TOP_UP
+  REFERRAL
+  SPEND
+  REFUND
+  MANUAL
+}
+
+model FiatPayment {
+  id                String             @id @default(cuid())
+  createdAt         DateTime           @default(now())
+  updatedAt         DateTime           @updatedAt
+  amount            BigInt // Amount in cents/smallest currency unit
+  currency          String // e.g., "usd"
+  status            FiatPaymentStatus
+  servicePaymentId  String             @unique // Stripe's payment_intent ID
+  serviceCustomerId String? // Stripe's customer ID
+  metadata          Json? // Additional payment metadata
+  creditTransaction CreditTransaction?
+  service           FiatPaymentService @default(STRIPE)
+}
+
+enum FiatPaymentService {
+  STRIPE
+}
+
+enum FiatPaymentStatus {
+  PENDING
+  SUCCEEDED
+  FAILED
+  REFUNDED
+  CANCELLED
+}
+
 model Job {
-  id                String    @id @default(cuid())
-  createdAt         DateTime  @default(now())
-  updatedAt         DateTime  @updatedAt
-  onChainIdentifier String    @unique
-  agentId           String
-  agent             Agent     @relation(fields: [agentId], references: [id])
-  userId            String
-  user              User      @relation(fields: [userId], references: [id])
-  status            JobStatus
-  input             String
-  output            String?
-  startedAt         DateTime  @default(now())
-  finishedAt        DateTime?
-  cost              BigInt
-  fee               BigInt    @default(0)
-  paymentTxId       String?
+  id                        String             @id @default(cuid())
+  createdAt                 DateTime           @default(now())
+  updatedAt                 DateTime           @updatedAt
+  agentJobId                String
+  paymentId                 String
+  onChainIdentifier         String             @unique
```

Comment:
Can you explain what the difference between `agentJobId` and `onChainIdentifier` is?

Reply:
yes one is public one is used to retrieve the results

Reply:
Can you explain it a bit more? How do I know which is for what?

As I understand I use the id of the job (onChainIdentifier) for `/status` call on the agentic service api. Then I receive a result, which is hashed and then I use another id to decode the result?!

Reply:
no you do not use those at all, there is the startJob function and then maybe a read to the job to see its status and results. All else is handled automatically